sdk/Context/Context.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OpenTelemetry\Sdk\Context;
+
+use OpenTelemetry\Trace as API;
+
+class Context implements API\Context
+{
+    /**
+     * @var ContextKey
+     */
+    private $key;
+
+    /**
+     * @var mixed
+     */
+    private $value;
+
+    /**
+     * @var Context
+     */
+    private $parent;
+
+    /**
+     * Private constructor. All instances should be created via withValue().
+     *
+     * This is heavily influenced by Golang's https://golang.org/pkg/context package
+     */
+    private function __construct(ContextKey $key, $value, ?Context $parent=null)
+    {
+        $this->key = $key;
+        $this->value = $value;
+        $this->parent = $parent;
+    }
+
+    /**
+     * @param ContextKey $key
+     * @param mixed $value
+     * @param Context|null $ctx
+     *
+     * @return Context
+     */
+    public static function withValue(ContextKey $key, $value, ?Context $ctx=null): API\Context
+    {
+        return new Context($key, $value, $ctx);
+    }
+
+    /**
+     * @param ContextKey $key
+     *
+     * @return mixed
+     */
+    public function value(ContextKey $key) {
+        if ($this->key === $key) {
+            return $this->value;
+        }
+        if (is_null($parent)) {
+            throw new ContextValueNotFoundException();
+        }
+        return $parent->value($key);
+    }
+}
```

Comment:
We also may want to add `GetContext`, `AttachContext`, and `DetatchContext` functions to this class?  Those are included in these tickets : 

https://github.com/open-telemetry/opentelemetry-php/issues/106
https://github.com/open-telemetry/opentelemetry-php/issues/107
https://github.com/open-telemetry/opentelemetry-php/issues/108

So perhaps you were leaving them for a separate PR.

Reply:
So, I was pondering those a bit. The spec describes those as behaviors that are optional and may be found useful depending on if the use of the OT provided Context implementation is "implicit" or "explicit" and I need to understand a bit more about that before I jump into writing those behaviors. They are like 4 lines of code each, but I think without understanding the practical usage, its hard to understand if we are providing boiler plate that matches the spec "just because" or if we are integrating the correct behavior at the right scope for good api ergonomics.

Reply:
I can see both sides.  If they are easy enough to create we could just create them, but it might not be worth having them in the codebase until we need them?  I don't hold any super strong opinions here.

Reply:
Yeah, and for stuff that's user-facing in the spec I 1000% agree. This is an implicit ecosystem need that is outlined by the spec but functionally is a little separate.